app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -325,12 +327,29 @@ protected function addDiscounts( $totals, &$cartSubmissionData, $quote = null )
                 // others as negative, which is the Magento default.
                 // Using the absolute value.
                 $discountAmount = (int) abs(round($amount * 100));
-
-                $cartSubmissionData['discounts'][] = array(
+                $data = array(
                     'amount'      => $discountAmount,
                     'description' => $totals[$discount]->getTitle(),
                     'type'        => 'fixed_amount',
                 );
+                if ($discount === 'discount' && $quote->getCouponCode()) {
+                    /////////////////////////////////////////////////////////////
+                    /// We want to get the apply the coupon code as a reference.
```

Comment:
why we need to set `reference` field? Is this just for our own reference?

Reply:
Hello, Kazuki.  For our own reference is a great reason in and of itself on several fronts.  Yes, for that reason, certainly.  In fact, if we have the ability store the reference to the coupon, or the shipping method, or an item, it may considered quite silly or a missed opportunity to omit it.

The primary reason that we are interested in the reference is that in pre-auth, we are tracking if values have changed and attempting to exactly report which element, if any, has changed.  For this, we look each element up, including coupons.  For certain, searching for a coupon identified by `primary key "39488420"` is much more reliable for look up than a title like `"Bob's Crazy Random Weekly Discount!!!"`